File: yandex_backend/7. Программист на пляже/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <limits>

using namespace std;

int main() {
    int t;
    cin >> t;

    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;

        vector<int> a(n);
        for (int j = 0; j < n; j++) {
            cin >> a[j];
        }

        sort(a.begin(), a.end()); 
        int minXor = numeric_limits<int>::max(); 

        for (int j = 1; j < n; j++) {
            int xorVal = a[j] ^ a[j - 1];
            minXor = min(minXor, xorVal);
        }

        cout << minXor << endl;
    }

    return 0;
}

File: yandex_backend/7. Программист на пляже/solution.py
import sys
from typing import List

def find_min_xor(arr: List[int]) -> None:
    arr.sort()
    min_xor: int = float('inf')
    for i in range(1, len(arr)):
        xor_val: int = arr[i] ^ arr[i-1]
        min_xor = min(min_xor, xor_val)
    print(min_xor)

def main() -> None:
    t: int = int(input())
    for _ in range(t):
        n: int = int(input())
        a: List[int] = list(map(int, input().split()))
        find_min_xor(a)

if __name__ == "__main__":
    main()